DigitalLearningSolutions.Data.Tests/Services/DelegateUploadFileServiceTests.cs

```diff
@@ -618,6 +622,104 @@ public void ProcessDelegateTable_calls_register_with_expected_values_when_welcom
             result.RegisteredCount.Should().Be(1);
         }
 
+        [Test]
+        [TestCase("-4")]
+        [TestCase("-3")]
+        [TestCase("-2")]
+        [TestCase("-1")]
+        public void ProcessDelegateTable_unsuccessful_register_does_not_update_supervisor_delegates(
+            string failureStatusCode
+        )
+        {
+            // Given
+            const string aliasId = "ALIAS";
+            var row = GetSampleDelegateDataRow(candidateNumber: string.Empty, aliasId: aliasId);
+            var table = CreateTableFromData(new[] { row });
+
+            A.CallTo(() => userDataService.GetDelegateUserByAliasId(aliasId, CentreId)).Returns(null);
+            A.CallTo(() => userService.IsDelegateEmailValidForCentre("email@test.com", CentreId)).Returns(true);
+            A.CallTo(() => registrationDataService.RegisterDelegateByCentre(A<DelegateRegistrationModel>._))
+                .Returns(failureStatusCode);

```

Comment:
Confused by the variable name a bit, does RegisterDelegateByCentre actually return a status code? I see the return variable in the method is called candidateNumber

Reply:
RegisterDelegateByCentre calls a stored procedure and returns what it returns. This can either by a candidate number if a delegate is registered successfully, or a non-zero status code if something goes wrong. These tests are specifically testing those error status codes prevent further processing.

Reply:
Understood, thanks for the clarification! :)